packages/govuk-frontend/src/govuk/components/task-list/template.njk

```diff
@@ -1,31 +1,30 @@
 {% from "../tag/macro.njk" import govukTag -%}
 
 {% set idPrefix = params.idPrefix if params.idPrefix else "task-list" %}
-<ul class="govuk-task-list {%- if params.classes %} {{ params.classes }}{% endif %}"{% for attribute, value in params.attributes %} {{attribute}}="{{value}}"{% endfor %}>
+<ul class="govuk-task-list {%- if params.classes %} {{ params.classes }}{% endif %}" {%- for attribute, value in params.attributes %} {{ attribute }}="{{ value }}"{% endfor %}>
   {% for item in params.items %}
-    {% if item.hint %}
-      {%- set hintId = idPrefix + "-" + loop.index + "-hint" -%}
-    {% endif %}
+    {%- set hintId = idPrefix + "-" + loop.index + "-hint" -%}
     {%- set statusId = idPrefix + "-" + loop.index + "-status" -%}
     <li class="govuk-task-list__item {%- if item.href %} govuk-task-list__item--with-link{% endif %}{%- if item.classes %} {{ item.classes }}{% endif %}">
       <div class="govuk-task-list__task-name-and-hint">
         {% if item.href %}
           <a class="govuk-link govuk-task-list__link {%- if item.title.classes %} {{ item.title.classes }}{% endif %}" href="{{ item.href }}" aria-describedby="{{ hintId + " " if item.hint }}{{ statusId }}">{{ item.title.html | safe if item.title.html else item.title.text }}</a>
         {% else %}
-          <div{% if item.title.classes %} class="{{ item.title.classes }}" {%- endif %}>{{ item.title.html | safe if item.title.html else item.title.text }}</div>
+          <div {%- if item.title.classes %} class="{{ item.title.classes }}"{% endif %}>{{ item.title.html | safe if item.title.html else item.title.text }}</div>
         {% endif %}
         {% if item.hint %}
           <div id="{{ hintId }}" class="govuk-task-list__task_hint">
             {{ item.hint.html | safe if item.hint.html else item.hint.text }}
           </div>
         {% endif %}
       </div>
-      <div class="govuk-task-list__status {{ item.status.classes }}" id="{{ statusId }}">
+      <div class="govuk-task-list__status {%- if item.status.classes %} {{ item.status.classes }}{% endif %}" id="{{ statusId }}">
         {% if item.status.tag %}
           {{ govukTag({
             text: item.status.tag.text,
             html: item.status.tag.html,
-            classes: item.status.tag.classes
+            classes: item.status.tag.classes,
+            attributes: item.status.tag.attributes
```

Comment:
Other nested components usually override `id` or `for` params

But we don't here. Should we just pass all the options in one?

```mjs
{{ govukTag(item.status.tag) }}
```

Reply:
That is a great shout 👍🏻